govcd/nsxt_network_context_profile.go

```diff
@@ -0,0 +1,77 @@
+package govcd
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+// GetAllNetworkContextProfiles retrieves a slice of types.NsxtNetworkContextProfile
+// This function requires at least a filter value for 'context_id' which can be one of:
+// * Org VDC ID - to get Network Context Profiles scoped for VDC
+// * Network provider ID - to get Network Context Profiles scoped for attached NSX-T environment
+// * VDC Group ID - to get Network Context Profiles scoped for attached NSX-T environment
+func GetAllNetworkContextProfiles(client *Client, queryParameters url.Values) ([]*types.NsxtNetworkContextProfile, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointNetworkContextProfiles
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	typeResponses := []*types.NsxtNetworkContextProfile{}
+	err = client.OpenApiGetAllItems(apiVersion, urlRef, queryParameters, &typeResponses, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return typeResponses, nil
+}
+
+// GetNetworkContextProfilesByScopeAndName retrieves a single NSX-T Network Context Profile by name
+// and context ID. All fields - name, scope and contextId are mandatory
+//
+// contextId is mandatory and can be one off:
+// * Org VDC ID - to get Network Context Profiles scoped for VDC
+// * Network provider ID - to get Network Context Profiles scoped for attached NSX-T environment
+// * VDC Group ID - to get Network Context Profiles scoped for attached NSX-T environment
+//
+// scope can be one off:
+// * SYSTEM
+// * PROVIDER
+// * TENANT
+func GetNetworkContextProfilesByNameScopeAndContext(client *Client, name, scope, contextId string) (*types.NsxtNetworkContextProfile, error) {
+	if name == "" || contextId == "" || scope == "" {
+		return nil, fmt.Errorf("error - 'name', 'scope' and 'contextId' must be specified")
```

Comment:
Does API error is good enough if the scope value isn't correct?

Reply:
It is not very bad IMO. Here is an example for "scope=HORN":
```
.. value *errors.errorString = &errors.errorString{s:"error getting all pages for endpoint https://bos1-vcloud-static-171-96.eng.vmware.com/cloudapi/1.0.0/networkContextProfiles: error in HTTP GET request: BAD_REQUEST - Cannot parse the supplied filter: scope==HORN"} ("error getting all pages for endpoint https://bos1-vcloud-static-171-96.eng.vmware.com/cloudapi/1.0.0/networkContextProfiles: error in HTTP GET request: BAD_REQUEST - Cannot parse the supplied filter: scope==HORN")
```

Reply:
Let's hope it will be enough :)